File: engine/engine-model/src/main/scala/tools/packet/foreigneffect/ShowPetLevelUp.scala
package tools.packet.foreigneffect

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowPetLevelUp(private var _characterId: Int, private var _index: Byte) extends PacketInput {
  def characterId: Int = _characterId

  def index: Byte = _index

  override def opcode(): SendOpcode = SendOpcode.SHOW_FOREIGN_EFFECT
}

File: engine/engine-model/src/main/java/tools/packet/foreigneffect/ShowPetLevelUp.java
package tools.packet.foreigneffect;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowPetLevelUp(Integer characterId, Byte index) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_FOREIGN_EFFECT;
   }
}